### DISCUSSION TITLE
gzip decompression seems not to work for default adapter

### DISCUSSION THREAD

**Discussion by wuarmin** [lostisland/faraday#1478] [Help]
Hey,

I need **gzip** decompression and the docs say, that this should work out of the box with the default adapter `Faraday::Adapter::NetHttp`.

![image](https://user-images.githubusercontent.com/9987680/210795838-9218599c-9d01-437e-80b3-a5e96f280c18.png)
But my response is not decompressed, although I set the `Accept-Encoding`-Header to `gzip, deflate` and the server responds properly:

```ruby
# response headers
{
  "server"            => "Apache-Coyote/1.1",
  "content-type"      => "text/xml;charset=UTF-8",
  "transfer-encoding" => "chunked",
  "content-encoding"  => "gzip",
  "vary"              => "Accept-Encoding",
  "date"              => "Thu, 05 Jan 2023 13:55:44 GMT"
}
```
What am I doing wrong?
thanks!

**Comment by iMacTia** [comment]
Faraday does not manually decompress responses, and instead it relies on the underlying adapter.
According to the [documentation](https://ruby-doc.org/stdlib-3.1.0/libdoc/net/http/rdoc/Net/HTTP.html#class-Net::HTTP-label-Compression), Net::HTTP automatically decompresses gzip responses, except in some exceptional cases.
There have also been cases in the past where some gem were monkey-patching Net::HTTP incorrectly causing this not to work (https://github.com/flexera-public/right_http_connection/pull/20).

It's hard to say what's wrong in your case, but my suggestion would be to test in isolation in order to identify the issue:
* Using plain Net::HTTP in a `irb` console, without Faraday or any other gem, does the decompression of that API response work?
  * YES: does it work using Faraday?
    * YES -> something specific to the project is affecting the decompression
    * NO -> There might actually be a bug in Faraday!
  * NO: something is wrong with Net::HTTP, maybe it's one of the exceptions mentioned in the doc
